R/qenv-join.R

```diff
@@ -1,12 +1,104 @@
 #' Join two `qenv` objects
 #'
-#' Combine two `qenv` objects by merging their environments and the code.
-#' Not all `qenv` objects can be combined:
-#' - if their environments contains objects of the same name but not identical
-#' - if `y` has unique code element placed before common element. This means that `y`
-#' in the environment of the `y` was evaluated some extra code before which can influence
-#' reproducibility
-#' - more cases to be done
+#' `join()` perform checks and merges two `qenv` objects into one `qenv` object.
```

Comment:
I would be tempted to add an extra line here:

`join()` perform checks and merges two `qenv` objects into one `qenv` object. Any common code at the start of the `qenv`s is only placed once at the start of the joined `qenv`. This allows consistent behaviour when joining `qenv`s which share common ancestor. See below for an example

And then in the examples section have something like:

```
common_q <- new_qenv(list2env(list(x = 1)), quote(x <- 1))
y_q <- eval_code(common_q, quote(y <- x * 2))
z_q <- eval_code(common_q, quote(z <- x * 3))
join_q <- join(y_q, z_q)

# get_code only has "x <-1" occurring once
get_code(join_q)
```

Reply:
I like it! Updated as suggested!